src/System.Security.Cryptography.Encoding/src/Internal/Cryptography/AsnFormatter.OSX.cs

```diff
@@ -0,0 +1,220 @@
+        {
+            // Because SubjectAlternativeName is a commonly parsed structure, we'll
+            // specifically format this one.  And we'll match the OpenSSL format, which
+            // includes not localizing any of the values (or respecting the multiLine boolean)
+            //
+            // The intent here is to be functionally equivalent to OpenSSL GENERAL_NAME_print.
+
+            // The end size of this string is hard to predict.
+            // * dNSName values have a tag that takes four characters to represent ("DNS:")
+            //   and then their payload is ASCII encoded (so one byte -> one char), so they
+            //   work out to be about equal (in chars) to their DER encoded length (in bytes).
+            // * iPAddress values have a tag that takes 11 characters ("IP Address:") and then
+            //   grow from 4 bytes to up to 15 characters for IPv4, or 16 bytes to 47 characters
+            //   for IPv6
+            //
+            // So use a List<string> and just Concat them all when we're done, and we reduce the
+            // number of times we copy the header values (vs pointers to the header values).
+            List<string> segments = new List<string>();
+
+            try
+            {
+                // SubjectAltName ::= GeneralNames
+                //
+                // IssuerAltName ::= GeneralNames
+                //
+                // GeneralNames ::= SEQUENCE SIZE (1..MAX) OF GeneralName
+                //
+                // GeneralName ::= CHOICE {
+                //   otherName                       [0]     OtherName,
+                //   rfc822Name                      [1]     IA5String,
+                //   dNSName                         [2]     IA5String,
+                //   x400Address                     [3]     ORAddress,
+                //   directoryName                   [4]     Name,
+                //   ediPartyName                    [5]     EDIPartyName,
+                //   uniformResourceIdentifier       [6]     IA5String,
+                //   iPAddress                       [7]     OCTET STRING,
+                //   registeredID                    [8]     OBJECT IDENTIFIER }
+                //
+                // OtherName::= SEQUENCE {
+                //   type - id    OBJECT IDENTIFIER,
+                //   value[0] EXPLICIT ANY DEFINED BY type - id }
+                DerSequenceReader altNameReader = new DerSequenceReader(rawData);
+
+                while (altNameReader.HasData)
+                {
+                    if (segments.Count != 0)
+                    {
+                        segments.Add(CommaSpace);
+                    }
+
+                    byte tag = altNameReader.PeekTag();
+
+                    if ((tag & DerSequenceReader.ContextSpecificTagFlag) == 0)
+                    {
+                        // All GeneralName values need the ContextSpecific flag.
+                        return null;
```

Comment:
I've noticed this `return null` and other error conditions here do not assert(false) or otherwise report invalid data in debug builds. Do we want to want to add such error checking, or is it common enough that it will just break debug builds?